BoxBrowseSDK/BoxBrowseSDK/BOXFolderViewController.m

```diff
@@ -400,44 +400,53 @@ - (void)fetchItemsWithCompletion:(void (^)(NSArray *items))completion
     folderRequest.SDKIdentifier = BOX_BROWSE_SDK_IDENTIFIER;
     folderRequest.SDKVersion = BOX_BROWSE_SDK_VERSION;
     folderRequest.requestAllFolderFields = YES;
-    [folderRequest performRequestWithCompletion:^(BOXFolder *folder, NSError *error) {
-        if (error) {
-            [self didFailToLoadItemsWithError:error];
-            self.navigationController.toolbarHidden = YES;
-        } else {
-            self.folder = folder;
-            [self fetchItemsInFolder:self.folder withCompletion:^(NSArray *items, NSError *error) {
-                if (error == nil) {
-                    self.title = self.folder.name;
-                    completion(items);
-                } else {
-                    self.navigationController.toolbarHidden = YES;
-                    [self didFailToLoadItemsWithError:error];
-                }
-                
-                if (self.tableView.visibleCells.count < 1) {
-                    [self switchToEmptyStateWithError:error];
-                    self.searchController.searchBar.hidden = YES;
-                } else {
-                    [self switchToNonEmptyView];
-                    self.searchController.searchBar.hidden = NO;
-                }
-            }];
+
+    void (^completionBlock)(NSArray *, BOOL, NSError *) = ^void(NSArray *items, BOOL fromCache, NSError *error) {
+        if (items.count > 0) {
+            [self switchToNonEmptyView];
+        } else if (!fromCache && self.tableView.visibleCells.count < 1) {
+            [self switchToEmptyStateWithError:error];
         }
+
+        if (completion) {
+            completion(items, fromCache, error);
+        }
+    };
+
+    void (^itemCacheBlock)(NSArray *, NSError *) = ^void(NSArray *items, NSError *error) {
+        completionBlock(items, YES, error);
+    };
+
+    void (^itemRefreshBlock)(NSArray *, NSError *) = ^void(NSArray *items, NSError *error) {
+        completionBlock(items, NO, error);
+    };
+
+    [folderRequest performRequestWithCached:^(BOXFolder *folder, NSError *error) {
```

Comment:
if error is about folder not found, or forbidden, e.g., we should reflect accordingly on the view (error.code == BOXContentSDKAPIErrorBadRequest || error.code == BOXContentSDKAPIErrorNotFound)
 and won't continue fetching its items